### ISSUE TITLE
Jest Error When Injecting AuthClientConfig Into Another Service

### ISSUE THREAD

**Issue by austinheinrichs** [auth0/auth0-angular#421]
### Description

I have another service (AppConfigService) that does dynamic configuration of the Auth0 client using an injected instance of AuthClientConfig. I am using the client config service indirectly to abstract some configurations across multiple projects. When mocking the AppConfigService after updating to @auth0/auth0-angular ^2, I noticed that I cannot plainly 'new up' an AuthClientConfig instance in the constructor of AppConfigService. Doing so results in the following error --> ReferenceError: Cannot access 'AuthConfigService' before initialization. Same error if I try providing a mock directly in the constructor. It appears to be due to the injection of AuthConfigService, but not sure how to get around it. Current workaround is creating a manual mock for AuthClientConfig and AuthConfigService in each project I'm using this in. Any guidance would be greatly appreciated.

![image](https://user-images.githubusercontent.com/12853049/216702273-e3ed6358-27be-4136-b7c3-be1dcfd69cac.png)
 

#### Example Simple Service
```
import { Injectable } from '@angular/core';
import { AuthClientConfig } from '@auth0/auth0-angular';

@Injectable({
  providedIn: 'root'
})
export class AppConfigService {

  constructor(private _authConfig: AuthClientConfig) {}

  /**
   * Set app configuration
   * @returns {Promise<void>}
   */
  public setAppConfiguration(): void {
    //Some logic here to set some standard configurations
    this._authConfig.set({
      //some configs
    });
  }
}
```
#### Example Simple Test
```
import { AuthClientConfig } from '@auth0/auth0-angular';

let service: AppConfigService;
describe('AppConfigService', () => {
  beforeEach(() => {
    jest.resetAllMocks();

    jest.spyOn(AuthClientConfig.prototype, 'set').mockReturnValue(void 0);

    service = new AppConfigService(new AuthClientConfig());

  });

  it('should be created', () => {
    expect(service).toBeTruthy();
  });

  it('should call AuthClientConfig set', ()=> {
    service.setAppConfiguration();
    expect(AuthClientConfig.prototype.set).toBeCalled();
  });
});

```
### Environment
- @angular/core@14.2.9
- @auth0/auth0-angular@2.0.0
- @angular-builders/jest@14.0.2
- jest@28.1.3

**Comment by frederikprijck** [comment]
Hey,

Can you try using version 2.0.1? This has been reported before, and fixed in 2.0.1.

Let me know if that doesnt solve it.

**Comment by austinheinrichs** [comment]
@frederikprijck That works, thank you! Sorry, I missed #413
